### DISCUSSION TITLE
[feat] automatically clear finished download from about:downloads

### DISCUSSION THREAD

**Discussion by Albirew** [onemen/TabMixPlus#320] [Ideas]
I believe there was an option like this in legacy TMP, where a finished download would clear itself from the download window.
I'd love if this feature could be brought back.
For now, the only way to clear download list is to either do it manually or to automatically clear the whole browser history when firefox closes...

**Comment by pyrates999** [comment]
That's there so you can then either open what you downloaded, or you can open the directory that the downloaded file was saved to.

try this:

Downloads are tied to browsing history, so to the best of my knowledge there is no setting which can separate their behavior other than **browser.download.clearHistoryOnDelete**. Setting this to 2 is supposed to clear downloads when the files are moved, renamed, or deleted. So if this really concerns you, try automating the renaming of downloads (e.g. Automator or Hazel on MacOS, for example).

source: https://www.reddit.com/r/firefox/comments/1d687pa/comment/l6quxkf

**Comment by onemen** [comment]
> I believe there was an option like this in legacy TMP, where a finished download would clear itself from the download window

Never was and never will be such an options in Tab Mix Plus
